src/Language/Source.php

```diff
@@ -28,7 +33,8 @@ public function __construct($body, $name = null)
         );
 
         $this->body = $body;
-        $this->length = mb_strlen($body, 'UTF-8');
+        $this->buf =  array_map('GraphQL\Utils::ord', preg_split('//u', $body, -1, PREG_SPLIT_NO_EMPTY));
```

Comment:
I've built a proof-of-concept version without mb or preg usage, which shows good results (see https://github.com/webonyx/graphql-php/issues/137#issuecomment-310839638). Will push it to new branch for v0.10.0 when ready.